### PR TITLE
chapter 14

### PR DESCRIPTION
**Pull request by chan227**

### CODE REVIEW

**Comment by chan227** [comment]
ready

**Comment by chan227** [comment]
ready

**Comment by chan227** [comment]
ready

**Comment by chan227** [comment]
hello

**Comment by chan227** [comment]
ready

**Comment by chan227** [comment]
ready

**Comment by chan227** [comment]
ready

**Comment by chan227** [comment]
ready

**Comment by chan227** [comment]
ready

**Comment by chan227** [comment]
ready

File: app/helpers/relationships_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module RelationshipsHelper
```
Comment (ngocvt-0484) [inline app/helpers/relationships_helper.rb]: remove it because you not use

File: app/models/user.rb [hunk 2]
```diff
@@ -70,7 +74,19 @@ def password_reset_expired?
   end
 
   def feed
-    microposts
+    Micropost.where("user_id IN (?) OR user_id = ?", following_ids, id)
```
Comment (ngocvt-0484) [inline app/models/user.rb]: where("user_id IN (?) OR user_id = ?", following_ids, id) => please use scope to define it

Comment (ngocvt-0484) [reply app/models/user.rb]: and use scope here

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -50,6 +50,20 @@ def destroy
     redirect_to users_path
   end
 
+  def following
+    @title = "Following"
+    @user = User.find_by id: params[:id]
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb]: please define a method to find user
and use before_action to call method

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -50,6 +50,20 @@ def destroy
...
+    @users = @user.following.paginate page: params[:page]
+    render "show_follow"
+  end
+
+  def followers
+    @title = "Followers"
+    @user = User.find_by id: params[:id]
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb]: please define a method to find user
and use before_action to call method

**Comment by chan227** [comment]
ready

File: app/models/relationship.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class Relationship < ApplicationRecord
+  belongs_to :follower, class_name: User.name
+  belongs_to :followed, class_name: User.name
+
+  validates :follower_id, presence: true
```
Comment (ngocvt-0484) [inline app/models/relationship.rb]:
```suggestion
  validates :follower_id, :followed_id, presence: true
```

**Comment by ngocvt-0484** [comment]
good job
please learn more about rails on https://guides.rubyonrails.org/

File: app/models/relationship.rb (hunk 5 above)
Comment (chan227) [reply app/models/relationship.rb]: Done

**Comment by chan227** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**